Source/Core/DiscIO/VolumeWii.cpp

```diff
@@ -170,42 +174,69 @@ bool VolumeWii::Read(u64 offset, u64 length, u8* buffer, const Partition& partit
   const PartitionDetails& partition_details = it->second;
 
   const u64 partition_data_offset = partition.offset + *partition_details.data_offset;
-  if (m_reader->SupportsReadWiiDecrypted(offset, length, partition_data_offset))
+  if (m_has_hashes && m_has_encryption &&
+      m_reader->SupportsReadWiiDecrypted(offset, length, partition_data_offset))
+  {
     return m_reader->ReadWiiDecrypted(offset, length, buffer, partition_data_offset);
+  }
 
-  if (!m_encrypted)
+  Common::AES::Context* aes_context = nullptr;
+  if (m_has_encryption)
   {
-    return m_reader->Read(partition.offset + *partition_details.data_offset + offset, length,
-                          buffer);
+    aes_context = partition_details.key->get();
+    if (!aes_context)
+      return false;
   }
```

Comment:
Would it make sense to keep this just-read branch for the !encrypted && !hash_hashes case? It would allow you to skip the checks of the different block sizes in the read/decrypt loop below.